Services/EnergyCostAlgorithm.cs

```diff
@@ -3,23 +3,32 @@ namespace Ultra_Saver;
 
 public class EnergyCostAlgorithm
 {
-    public double totalEnergy { get; set; } = 0;
-    public double totalEnergyCost { get; set;} = 0;
-    private float costPerkWh = 0;
+    public float TotalEnergy { get; set; } = 0;
 
     // powerScale is a proportion of power used for the specified time; scales 1 to 10; ex.: medium heat, 3/10 of power
-    private void ElectricPower (int appliancePower, short powerScale, int timeInMin){
-        totalEnergy += appliancePower * powerScale * timeInMin / 600_000;
-        totalEnergyCost = Math.Round (totalEnergy * costPerkWh, 2);
+    // if User's appliance power efficiancy is 80% then ApplianceEfficiency = 0.8
+    public void ElectricPower (short AppliancePower, byte PowerScale, short TimeInMin, float ApplianceEfficiency){
```

Comment:
There is no need to optimize memory usage to such a level. Moreover, albeit not very likely, it is not completely unreasonable for appliancePower to be > ~32kWh

Reply:
In my point of view, optimizing memory is always great. Things such as AppliancePower could be declared as int, however PowerScale, which value will always be 1-10, can safely be declared as byte type.